metaserver/src/main/scala/scala/meta/languageserver/Compiler.scala

```diff
@@ -47,41 +54,97 @@ class Compiler(
         )
       }
 
+  def signatureHelp(
+      path: AbsolutePath,
+      line: Int,
+      column: Int
+  ): SignatureHelp = {
+    getCompiler(path, line, column - 1).fold(noSignatureHelp) {
+      case (compiler, position) =>
+        logger.info(s"Signature help at $path:$line:$column")
+        // Find the last leading open paren to get the method symbol
+        // of this argument list. Note, this may still cause false positives
+        // in cases like `foo(bar(), <cursor>)` since we will find the
+        // symbol of `bar` when we want to match against `foo`.
+        // Related https://github.com/scalameta/language-server/issues/52
+        val lastParenOffset =
+          position.source.content.lastIndexOf('(', position.point)
+        if (lastParenOffset < 0) {
+          // bail if we can't find open paren to prevent false positives.
+          noSignatureHelp
+        } else {
+          val lastParenPosition = position.withPoint(lastParenOffset)
+          val signatureInformations = for {
+            member <- compiler
+              .completionsAt(lastParenPosition)
+              .matchingResults()
+              .distinct
+            if member.sym.isMethod
+          } yield {
+            val sym = member.sym.asMethod
+            val parameterInfos = sym.paramLists.headOption.map { params =>
+              params.map { param =>
+                ParameterInformation(
+                  label = s"${param.nameString}: ${param.info.toLongString}",
+                  documentation = None
+                )
+              }
+            }
+            SignatureInformation(
+              label = s"${sym.nameString}${sym.info.toLongString}",
```

Comment:
I used this just from what already was around. But is there a better way to get separately parameters list and return type?

<img width="209" alt="screen shot 2017-11-23 at 13 41 40" src="https://user-images.githubusercontent.com/766656/33181089-42e0e168-d06f-11e7-8759-4c026867a861.png">